ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactionsTest.java

```diff
@@ -132,15 +120,29 @@ public void shouldPrioritizeLocalTransaction() {
   }
 
   @Test
-  public void shouldStartDroppingLocalTransactionsWhenPoolIsFullOfLocalTransactions() {
-    final Transaction firstLocalTransaction = createTransaction(0);
-    transactions.addLocalTransaction(firstLocalTransaction);
+  public void shouldAcceptLocalTransactionIfFullAndTxBetterPriced() {
+    final int startGasPrice = 1;
+    int gasPrice = startGasPrice;
+    for (int i = 1; i <= MAX_TRANSACTIONS; i++, gasPrice++) {
+      transactions.addLocalTransaction(createTransaction(i, Wei.of(gasPrice)));
+    }
+    assertThat(transactions.size()).isEqualTo(MAX_TRANSACTIONS);
+    assertTrue(
+        transactions.addLocalTransaction(
+            createTransaction(MAX_TRANSACTIONS + 1, Wei.of(startGasPrice + 1))));
+  }
 
-    for (int i = 1; i <= MAX_TRANSACTIONS; i++) {
-      transactions.addLocalTransaction(createTransaction(i));
+  @Test
+  public void shouldNotAcceptLocalTransactionIfFullAndUnderpriced() {
```

Comment:
I think we will need more test cases, so a parameterized unit test may be in order.

All of these assume full local pools.


* For a queue with a single account and nonces 1,2,3 when 2 is the lowest price, and a new TX at 4 comes in higher than 2, do we drop 2 or reject?
* For a queue with a single account and nonces 1,2,3 when 2 is the lowest price, and a new TX on a different account higher than 2 comes in, do we  drop 2 or reject?  if we drop 2 do we drop 3?
* account A with nonce 1, gas 1, account B with nonce 3, gas 3.  new TX with nonce 2, gas 2, we should drop TX A1.  If I read it correctly we will reject.
* account A with nonce 1, gas 3, account B with nonce 3, gas 3.  new TX with nonce 2, gas 2, we should reject.